src/app/services/test-stations/test-stations.service.spec.ts

```diff
@@ -0,0 +1,85 @@
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { TestBed } from '@angular/core/testing';
+import { TestStation } from '@models/test-station.model';
+import { provideMockStore } from '@ngrx/store/testing';
+import { environment } from '../../../environments/environment';
+import { initialAppState } from '../../store';
+import { TestStationsService } from './test-stations.service';
+
+describe('TechnicalRecordService', () => {
+  let service: TestStationsService;
+  let httpTestingController: HttpTestingController;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [TestStationsService, provideMockStore({ initialState: initialAppState })]
+    });
+
+    httpTestingController = TestBed.inject(HttpTestingController);
+    service = TestBed.inject(TestStationsService);
+  });
+
+  afterEach(() => {
+    // After every test, assert that there are no more pending requests.
+    httpTestingController.verify();
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  describe('fetchTestStations', () => {
+    it('should get an array of matching results', () => {
+      const expectedResult = [{ testStationName: 'Some Name' } as TestStation];
+      service.fetchTestStations().subscribe(response => expect(response).toEqual(expectedResult));
+
+      // Check for correct requests: should have made one request to search from expected URL
+      const req = httpTestingController.expectOne(`${environment.VTM_API_URI}/test-stations/`);
+      expect(req.request.method).toEqual('GET');
+
+      // Provide each request with a mock response
+      req.flush(expectedResult);
+    });
+
+    it('should handle errors', () => {
```

Comment:
This test and the one one line 72 seem to not be doing what they are supposed to. Why would you expect to have a test station in the data if the server responds with a 500? I think you need to use the `done()` function inside the subscribe. Also these unit tests are both failing on the branch, if you look in the `checks` tab on the PR.

Reply:
Suggestion applied. Good spot - thanks!